pykrige/core.py

```diff
@@ -266,119 +304,105 @@ def initialize_variogram_model(x, y, z, variogram_model, variogram_model_paramet
+    if weight:
+        drange = np.amax(x) - np.amin(x)
+        k = 2.1972 / (0.1 * drange)
+        x0 = 0.7 * drange + np.amin(x)
+        weights = 1. / (1. + np.exp(-k * (x0 - x)))
+        weights /= np.sum(weights)
+        resid = (variogram_function(params, x) - y) * weights
     else:
-        if variogram_model == 'custom':
-            raise ValueError("Variogram parameters must be specified when implementing custom variogram model.")
-        else:
-            variogram_model_parameters = calculate_variogram_model(lags, semivariance, variogram_model,
-                                                                   variogram_function, weight)
-
-    return lags, semivariance, variogram_model_parameters
+        resid = variogram_function(params, x) - y
 
+    return resid
 
-def variogram_function_error(params, x, y, variogram_function, weight):
-    """Function used to in fitting of variogram model.
-    Returns RMSE between calculated fit and actual data."""
 
-    diff = variogram_function(params, x) - y
-
-    if weight:
-        weights = np.arange(x.size, 0.0, -1.0)
-        weights /= np.sum(weights)
-        rmse = np.sqrt(np.average(diff**2, weights=weights))
-    else:
-        rmse = np.sqrt(np.mean(diff**2))
+def _calculate_variogram_model(lags, semivariance, variogram_model, variogram_function, weight):
+    """Function that fits a variogram model when parameters are not specified.
 
-    return rmse
+    Parameters
+    ----------
+    lags: 1d array
+        binned lags/distances to use for variogram model parameter estimation
+    semivariance: 1d array
+        binned/averaged experimental semivariances to use for variogram model parameter estimation
+    variogram_model: str/unicode
+        specified variogram model to use for parameter estimation
+    variogram_function: callable
+        the actual funtion that evaluates the model variogram
+    weight: bool
+        flag for implementing the crude weighting routine, used in order to fit smaller lags better
+        this is passed on to the residual calculation function, where weighting is actually applied...
 
+    Returns
+    -------
+    res: list
+        list of estimated variogram model parameters
 
-def calculate_variogram_model(lags, semivariance, variogram_model, variogram_function, weight):
-    """Function that fits a variogram model when parameters are not specified."""
+    NOTE that the estimation routine works in terms of the partial sill (psill = sill - nugget) --
+    setting bounds such that psill > 0 ensures that the sill will always be greater than the nugget...
+    """
 
     if variogram_model == 'linear':
-        x0 = [(np.amax(semivariance) - np.amin(semivariance))/(np.amax(lags) - np.amin(lags)),
-              np.amin(semivariance)]
-        bnds = ((0.0, 1000000000.0), (0.0, np.amax(semivariance)))
+        x0 = [(np.amax(semivariance) - np.amin(semivariance))/(np.amax(lags) - np.amin(lags)), np.amin(semivariance)]
```

Comment:
Generally speaking the previous formatting aimed at keeping the lines below 80 characters (I imagine). Not sure if we want that...